src/main/java/org/sonar/plugins/stash/client/StashClient.java

```diff
@@ -456,4 +459,33 @@ AsyncHttpClient createHttpClient(String sonarQubeVersion) {
             .build()
     );
   }
+
+  private Collection<JsonObject> performPagedRequest(BoundRequestBuilder requestBuilder,
+      JsonObject body,
+      int expectedStatusCode,
+      String errorMessage) throws StashClientException {
+
+    Collection<JsonObject> result = new ArrayList<>();
+
+    boolean doneLastPage = false;
+    int nextPageStart = 0;
+    // FIXME size/limit support
+
+    while (!doneLastPage) {
+      if (nextPageStart > 0) {
+        requestBuilder.addQueryParam("start", String.valueOf(nextPageStart));

```

Comment:
This will add an additional parameter "start" and not replace the previous one. With more activities you will end up with link params:
?start=25&start=50&start50&start-50... 
which will at the end fail.

Reply:
Adding
requestBuilder.resetQuery();
just before 
requestBuilder.addQueryParam(...
should fix the issue.

Reply:
@marekjagielski Good point. thanks!